_job-scheduler-plugin/index.md

```diff
@@ -0,0 +1,130 @@
+    private Schedule schedule;
+    private String indexToWatch;
+    private Long lockDurationSeconds;
+    private Double jitter;
+```
+
+Next, configure the request parameters you would like your plugin to use with Job Scheduler. These will be based on the variables that are declared when configuring your plugin. The sample below shows the request parameters you set when building your plugin:
+
+```java
+public SampleJobParameter(String id, String name, String indexToWatch, Schedule schedule, Long lockDurationSeconds, Double jitter) {
+        this.jobName = name;
+        this.indexToWatch = indexToWatch;
+        this.schedule = schedule;
+
+        Instant now = Instant.now();
+        this.isEnabled = true;
+        this.enabledTime = now;
+        this.lastUpdateTime = now;
+        this.lockDurationSeconds = lockDurationSeconds;
+        this.jitter = jitter;
+    }
+
+    @Override
+    public String getName() {
+        return this.jobName;
+    }
+
+    @Override
+    public Instant getLastUpdateTime() {
+        return this.lastUpdateTime;
+    }
+
+    @Override
+    public Instant getEnabledTime() {
+        return this.enabledTime;
+    }
+
+    @Override
+    public Schedule getSchedule() {
+        return this.schedule;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return this.isEnabled;
+    }
+
+    @Override
+    public Long getLockDurationSeconds() {
+        return this.lockDurationSeconds;
+    }
+
+    @Override public Double getJitter() {
+        return jitter;
+    }
+```
+
+The following table describes the request parameters configured in the previous sample. All the request parameters shown are required.
+
+| Field | Data type | Description |
+:--- | :--- | :---
+| getName | String | Returns the name of the job. |
+| getLastUpdateTime | Time unit | Returns the last time the job ran. |
+| getEnabledTime | Time unit | Returns the time that the job was enabled. |
+| getSchedule | Unix cron | Returns the schedule that the job runs on formatted in Unix cron syntax. |
+| isEnabled | Boolean | Whether or not the job is enabled. |
+| getLockDurationSeconds | Integer | Returns the duration of time that the job is locked. |
+| getJitter | Integer | Returns the defined jitter value. |
+
+The logic used by your job should be defined by a class extended from `ScheduledJobRunner` in the `SampleJobParameter.java` sample file, such as `SampleJobRunner`. While the job is running, there is a locking mechanism you can use to prevent other nodes from running the same job. [Acquire](https://github.com/opensearch-project/job-scheduler/blob/main/sample-extension-plugin/src/main/java/org/opensearch/jobscheduler/sampleextension/SampleJobRunner.java#L96) the lock. Then, make sure to release the lock before the [job finishes](https://github.com/opensearch-project/job-scheduler/blob/main/sample-extension-plugin/src/main/java/org/opensearch/jobscheduler/sampleextension/SampleJobRunner.java#L116).
```

Comment:
```suggestion
The logic used by your job should be defined by a class extended from `ScheduledJobRunner` in the `SampleJobParameter.java` sample file, such as `SampleJobRunner`. While the job is running, there is a locking mechanism you can use to prevent other nodes from running the same job. First [acquire](https://github.com/opensearch-project/job-scheduler/blob/main/sample-extension-plugin/src/main/java/org/opensearch/jobscheduler/sampleextension/SampleJobRunner.java#L96) the lock. Then make sure to release the lock before the [job finishes](https://github.com/opensearch-project/job-scheduler/blob/main/sample-extension-plugin/src/main/java/org/opensearch/jobscheduler/sampleextension/SampleJobRunner.java#L116).
```